native/SampleApps/AuthFlowTester/AuthFlowTesterUITests/PageObjects/LoginPageObject.swift

```diff
@@ -0,0 +1,303 @@
+ are permitted provided that the following conditions are met:
+ * Redistributions of source code must retain the above copyright notice, this list of conditions
+ and the following disclaimer.
+ * Redistributions in binary form must reproduce the above copyright notice, this list of
+ conditions and the following disclaimer in the documentation and/or other materials provided
+ with the distribution.
+ * Neither the name of salesforce.com, inc. nor the names of its contributors may be used to
+ endorse or promote products derived from this software without specific prior written
+ permission of salesforce.com, inc.
+ 
+ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR
+ IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+ CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY
+ WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import Foundation
+import XCTest
+import SalesforceSDKCore
+
+/// Page object for interacting with the Salesforce login screen during UI tests.
+/// Provides methods to configure login servers, login options and perform user authentication.
+class LoginPageObject {
+    let app: XCUIApplication
+    let timeout: double_t = 5
+    
+    init(testApp: XCUIApplication) {
+        app = testApp
+    }
+    
+    func isShowing() -> Bool {
+        return loginNavigationBar().waitForExistence(timeout: timeout)
+    }
+    
+    func configureLoginHost(host: String) -> Void {
+        tap(settingsButton())
+        tap(changeServerButton())
+        
+        if (hasHost(host: host)) {
+            // Select host if it exists already
+            tap(hostRow(host: host))
+        } else {
+            // Add host if it does not exist
+            tap(addConnectionButton())
+            setTextField(hostInputField(), value: host)
+            tap(doneOnAddConnectionButton())
+        }
+    }
+    
+    func performLogin(username: String, password: String) {
+        setTextField(usernameField(), value: username)
+        setTextField(passwordField(), value: password)
+        tap(loginButton())
+        tapIfPresent(allowButton())
+    }
+    
+    func configureLoginOptions(
+        staticAppConfig: AppConfig?,
+        staticScopes: String,
+        dynamicAppConfig: AppConfig?,
+        dynamicScopes: String,
+        useWebServerFlow: Bool,
+        useHybridFlow: Bool,
+        supportWelcomeDiscovery: Bool,
```

Comment:
I love the use of [trailing commas](https://github.com/swiftlang/swift-evolution/blob/main/proposals/0439-trailing-comma-lists.md), I didn't know that was in Swift .